### DISCUSSION TITLE
2024-08-19 회의록

### DISCUSSION THREAD

**Discussion by Captain-Kim** [nbc-final-a10/TripBuddy#287] [회의록]
#### > 튜터 모의 면접

<table>
    <tr>
        <td><strong>시간</strong></td>
        <td><strong>이름</strong></td>
    </tr>
    <tr>
        <td>12:20 ~ 13:00</td>
        <td>오은</td>
    </tr>
    <tr>
        <td>17:20 ~ 18:00</td>
        <td>김병준</td>
    </tr>
</table>

#### > 참여기업 튜터 모의 면접

<table>
    <tr>
        <td><strong>팀원</strong></td>
        <td><strong>날짜</strong></td>
        <td><strong>시간</strong></td>
        <td><strong>튜터</strong></td>
    </tr>
    <tr>
        <td>오은</td>
        <td>24-08-19</td>
        <td>20 : 20 ~ 20: 40</td>
        <td>진호인</td>
    </tr>
    <tr>
        <td>김병준</td>
        <td>24-08-19</td>
        <td>20 : 40 ~ 21 :00</td>
        <td>진호인</td>
    </tr>
    <tr>
        <td>김민지</td>
        <td>24-08-19</td>
        <td>20 : 00 ~ 20 : 20</td>
        <td>김건희</td>
    </tr>
    <tr>
        <td>곽현정</td>
        <td>24-08-19</td>
        <td>20 : 20 ~ 20: 40</td>
        <td>김건희</td>
    </tr>
    <tr>
        <td>최유상</td>
        <td>24-08-19</td>
        <td>20 : 40 ~ 21 : 00</td>
        <td>김건희</td>
    </tr>

</table>

#### > 준비사항

아래 내용을 준비하셔야 하고, 개인별 제출 문서 내용과 같습니다.

**1. 기술 스택 이해**
- 내가 사용한 기술 스택에 대해 깊이 있게 이해하고 있는지 확인하세요.
- 주요 기능과 그 기술을 선택한 이유를 설명할 수 있어야 합니다.

**2. 프로젝트 설명**
- 프로젝트의 목표, 역할, 사용 기술, 결과 등을 간단명료하게 설명할 수 있도록 준비하세요.
- 예상 질문에 대해 구체적인 답변을 준비하세요.

**3. 문제 해결 사례**
- 프로젝트 진행 중 겪었던 문제와 그 해결 과정을 설명할 수 있도록 하세요.
- 문제 해결 능력을 강조할 수 있는 실제 사례를 준비하세요.

**4. 자기소개**
- 간결하고 명확하게 자신을 소개할 수 있도록 연습하세요.
- 직무와 관련된 경험과 강점을 부각시키세요.

**5. 커뮤니케이션**
- 면접관의 질문을 명확히 이해하고, 정확히 답변하는 연습을 하세요.
- 논리적으로 생각하고 답변하는 능력을 기르세요.

### 🔥 개발 관련
- 검색기능 상황 체크

### 🔥 기타 내용

**- [병준]** : 노션 브로셔 깃허브로 옮길 예정 (UT 결과보고서 포함)
- 현정님 병준님 결과보고서 / 최종 PPT 개발 외 티켓 변경 내용
현정님 그리고 개발 외 티켓 변경을 좀 제안드려볼까 하는데요,

병준 : UT 결과보고서
현정님 : 최종 PPT

병준 : 최종 PPT
현정님 : UT 결과보고서

어차피 리드미나 PPT나 내용이 같은데 시각화되어있냐 글로 되어있냐 차이일 뿐이라
리드미 자체를 글로 나열하는 것보다 시각화된 PPT 자료로 리드미를 대체하는 것이
조금 더 신선할 것 같다는 생각을 했는데요
이렇게 한 번 제안드려보고자 하는데 편하게 의견 주시면 감사하겠습니다

### 🔥 타임라인 점검

- **최종 프로젝트 발표 이전 개발 외 티켓 점검**
  - **공통** : 더미 데이터 작성
    - 진짜 서비스인 것처럼. 짧은 글 X 긴 글 O, 채팅도 실제로 진행, 계정도 실제 유저처럼
  - **오은** : 발표 대본 작성 및 발표 예행 연습
  - **김병준** : 리드미/위키 작성, UT 결과보고서 작성
  - **김민지** : 시연 영상 촬영
  - **곽현정** : PPT 제작 -> UT 결과보고서와 변경 제안 (병준)
- **최종주차 타임라인**
  - **8월 19일 월요일 아침** -> 각 기능 담당자 시연하며 설명 및 (치명적인) 버그 리뷰 및 픽스
    - 배포 버그 확인되어 은님 픽스 후 12:00~ 진행
  - **8월 19일 월요일 저녁까지** -> 개인별 문서 작성 후 팀 공유 :: 문서 서식은 댓글 참고 (최대한 길고 자세하게)
  - **8월 20일 화요일 아침까지** -> 개인별 비개발 담당 티켓 제작 완료
  - **8월 20일 화요일 저녁회의** -> 최종 공유
  - **8월 21일 수요일** -> 최종발표

### 🔥 부제목

- 내용

## 🍀 공통 트러블 슈팅

- 내용

// 이미지 테이블
<table>
  <tr>
    <th>이미지 1번 제목</th>
    <th>이미지 2번 제목</th>
  </tr>
  <tr>
    <td><img src="이미지 주소" width="300" height="300"/></td>
    <td><img src="이미지 주소" width="300" height="300"/></td>
  </tr>
</table>

**Comment by Captain-Kim** [comment]
## 개인별 회고록 (김병준)

이 댓글에 대댓글로 부탁드립니다.
한 달 동안 고생 많이들 하셨으니, 기록보드나 회의록 참고하셔서
포폴 작성 하신다고 생각하고 최대한 자세하게 서술 부탁드립니다.

발표 때 생략 되더라도 리드미, PPT에는 전부 싣겠습니다.

- 어필하고 싶은 부분 부끄러워 마시고 고생하신 만큼 제대로 PR 해서 다 적어 주시고,
- 공통으로 한 기능도 본인이 참여한 내용을 강조해서 전부 적어주세요.
- 트러블 슈팅은 기술적 의사 결정 과정과 사고의 흐름이 잘 보이도록 해결 과정을 상세히 서술해주세요.

### 🌈 맡은 기능

1) 마이페이지
2) 글쓰기
3) 소셜 기능
- 팔로우, 팔로잉
- 여정(파티) 참여하기, 나가기, 삭제하기
4) 온도지수

### 🌈 사용한 라이브러리와 이유

- react-range : input 태그로 구현 시도 하였으나... 이러저러한 이유로...
- next-ui : 캘린더 컴포넌트를 직접 개발하기엔 데드라인을 지키지 못할 것 같아 사용함

### 🌈 기술적으로 강조할 부분

1) 퍼넬 패턴
2) 재사용 가능한 컴포넌트 및 커스텀 훅 설계

### 🌈 기술적 의사 결정 과정/사유

### 🌈 트러블 슈팅과 해결 과정

### 🌈 개선 과제

### 🌈 팀에 대한 평가

- 10점 만점 중 9점
- 이유는... 잘 했다고 생각하는 부분은 ... 아쉬웠던 부분은 ...

### 🌈 스스로에 대한 평가

- 10점 만점 중 8점
- 이유는... 잘 했다고 생각하는 부분은 ... 아쉬웠던 부분은 ...

### 🌈 부트캠프를 마치며 하고 싶은 말

**Reply by Kyara0324** [reply]
## 개인별 회고록 (최유상)

### 🌈 맡은 기능
- 튜토리얼 개발: 이 프로젝트에서 사용자가 애플리케이션의 주요 기능을 이해할 수 있도록 돕는 튜토리얼 기능을 개발했습니다.

### 🌈 사용한 라이브러리와 이유
- React.js: UI를 구성하기 위한 기본 프레임워크로 사용했습니다.
- Next.js: 서버사이드 렌더링과 라우팅을 쉽게 구현하기 위해 사용했습니다.
- Tailwind CSS: 튜토리얼 UI를 빠르고 효율적으로 스타일링하기 위해 사용했습니다.

### 🌈 기술적으로 강조할 부분
- 퍼넬 패턴: 튜토리얼의 각 단계가 점진적으로 좁아지는 퍼넬 구조를 따라, 사용자가 핵심 기능을 단계적으로 이해할 수 있도록 구성했습니다.
- 튜토리얼 UI: 직관적이고 사용자 친화적인 UI를 구현하기 위해 다양한 애니메이션 효과와 사용자 상호작용을 고려했습니다.

### 🌈 기술적 의사 결정 과정/사유
- Tailwind CSS 채택: CSS-in-JS 방식보다는 간단하고 신속하게 스타일링 작업을 할 수 있는 Tailwind CSS를 사용했습니다.

### 🌈 트러블 슈팅과 해결 과정
- 상태 관리 문제: 튜토리얼 단계가 리프레시될 때 초기화되는 문제를 발견했습니다. 이를 해결하기 위해 상태 관리를 개선하고 튜토리얼 진행 상황을 로컬 스토리지에 저장하여 문제를 해결했습니다.
- UI 렌더링 이슈: 일부 브라우저에서 튜토리얼 UI가 제대로 렌더링되지 않는 문제를 발견했고, 이 문제는 Tailwind CSS의 특정 설정을 조정하여 해결했습니다.

### 🌈 개선 과제
- 튜토리얼 커스터마이제이션: 사용자가 튜토리얼 단계를 다시 볼 수 있도록 하는 기능을 추가할 필요가 있습니다.
- 반응형 디자인 개선: 다양한 디바이스에서 일관된 사용자 경험을 제공하기 위해 반응형 디자인을 더 강화할 필요가 있습니다.

### 🌈 팀에 대한 평가
- 10점 만점 중 10점
- 잘 했다고 생각하는 부분: 팀원들이 각자의 역할을 잘 수행하며, 서로의 작업을 적극적으로 지원한 점이 좋았습니다.
- 아쉬웠던 부분: 팀적으로는 완벽했으나 개인적으로는 리더님과 부리더님을 좀 더 많이 도와드리지못해 아쉬운 마음입니다.

### 🌈 스스로에 대한 평가
- 10점 만점 중 5점
- 잘 했다고 생각하는 부분: 맡은 기능을 시간 내에 구현하고 끝까지 포기하지 않은부분 입니다.
- 아쉬웠던 부분: 이왕 하는거면 더욱 많은 기능을 맡아서 열심히 최선을 다해서 했으면 좋았겠다 라는 생각이 듭니다.

### 🌈 부트캠프를 마치며 하고 싶은 말
- 스파르타 코딩클럽을 통해 많은 것을 배웠고, 특히 팀 작업의 중요성을 실감했습니다. 앞으로도 계속해서 성장하며, 더 나은 사람이 되고자 노력하겠습니다. 팀원분들 한달간 정말 고생 많으셨습니다.

**Reply by eunohhh** [reply]
## 개인별 회고록 (오은)

### 🌈 맡은 기능

1) 인증 / 인가
2) 회원정보 수정
3) 실시간 알림 기능
4) 온도지수 관련 기능
5) 스토리 관련 기능
6) 이미지 자동생성 기능

### 🌈 사용한 라이브러리와 이유

- cva : 재사용성 높은 공용 컴포넌트를 위해 사용함
- openai : 이미지 자동생성을 위해 사용함
- react-intersection-observer : 무한 스크롤 구현을 위해 사용함
- sharp : 서버측에서 이미지를 webp 로 변환하기 위해 사용함
- react-draggable : 요소 드래그를 구현하기 위해 사용함

### 🌈 기술적으로 강조할 부분

1) 재사용 가능한 컴포넌트 및 커스텀 훅 설계
2) 서버측 프리페치 쿼리
3) 인증/인가 레이어를 context api 로 구현
3) 수파베이스 리얼타임을 활용한 알림 기능을 context api 로 구현

### 🌈 기술적 의사 결정 과정/사유
- nextjs 의 장점을 활용하기 위해 가능한 경우 최대한 서버측에서 프리페치 쿼리 하여, 초기 데이터가 있는 상태로 클라이언트 컴포넌트가 시작되게 하였습니다.
- 서버로부터 prop 을 받아 initialState 를 설정할 수 있는 context api 를 활용하여 전역상태 관리 라이브러리를 사용하지 않고도 상태관리를 효율적으로 할 수 있었습니다.
- 아토믹 패턴을 최대한 활용하여 효율적인 컴포넌트 관리를 하려고 하였습니다. 이 과정에서 cva를 사용하여 최대한 쉽게 재사용 가능한 컴포넌트를 만들려고 하였습니다.

### 🌈 트러블 슈팅과 해결 과정
- pwa 를 구현함에 있어 ios 에서는 안내를 보여주는 방법밖에는 없다는 것을 확인했고, 해결방법이 없음에 따라 제외하게 되었습니다.
- supabase 인증에서 발생하는 Auth session missing! 에러에 대처하기 위해, 이 부분을 라우트 핸들러에서 예외 처리하여 해결할 수 있었습니다.
- 모바일에서 당겨서 새로고침 동작을 막아야 하는 상황이 있었는데 useEffect 에서 documentElement 에 선택적으로 overscrollBehavior='none'을 주어서 해결했습니다.

### 🌈 개선 과제
- 라우트핸들러 사용법, fetch 사용법, 탠스택쿼리 사용법 등에서 통일감 있게 진행되지 못한 부분이 많습니다. 개선이 필요합니다.
- 아토믹 패턴을 가능케 하는 컴포넌트 분리에 있어 팀원간에 컨벤션 통일이 정확히 이루어지지 못한 것 같습니다. 개선이 필요합니다.
- 코드 스플리팅을 통한 사이트 속도 개선이 필요해보입니다.
- seo, web vital 부분에서 부족한 부분이 많습니다. 역시 개선이 필요해보입니다.

### 🌈 팀에 대한 평가

- 10점 만점 중 9점
- 이유는 소통이 잘 되었고 긴밀하게 협업하면서 완성까지 다함께 달려온 것 같습니다.

### 🌈 스스로에 대한 평가

- 10점 만점 중 5점
- 컨벤션, 아토믹 패턴, 퍼널 패턴, 탠스택쿼리 사용, 라이브러리 사용 모든 측면에서 프로젝트가 통일감있게 진행되도록 컨트롤 하지 못한 것 같아서 아쉽습니다.

### 🌈 부트캠프를 마치며 하고 싶은 말
- 모두 그동안 빡빡한 일정 속에 너무나 수고 많으셨습니다.

**Reply by hyunjeongkwak** [reply]
## 개인별 회고록 (곽현정)

### 🌈 맡은 기능

- 다중 옵션 검색 기능

### 🌈 사용한 라이브러리와 이유

- react-range : 멀티 레인지 슬라이더 구현이 어려워, 라이브러리 이용
- next-ui : 캘린더를 직접 개발하기엔 어려움이 있어 사용

### 🌈 기술적으로 강조할 부분

- 지역, 국가 선택을 context api로 구현
- applyFilters 함수에서 필터링과 정렬 로직을 별도의 유틸리티 함수로 분리하여 코드의 가독성을 높이고 유지보수를 용이하게 하고자 했습니다.

### 🌈 기술적 의사 결정 과정/사유

- 미디어 쿼리를 사용하여 isXL 상태를 통해 화면 크기에 따라 다른 레이아웃을 적용하여 반응형 디자인을 구현했습니다.
- 콜백 함수를 메모이제이션하여 성능을 최적화하고, 불필요한 재렌더링을 방지하고자 했습니다.
- useEffect와 fetchFilteredTrips를 통해 비동기로 데이터를 가져오고, 필터링된 결과를 상태에 설정하여 동적으로 UI를 업데이트하고자 했습니다.

### 🌈 트러블 슈팅과 해결 과정

- 무한 렌더링 문제: 상태 업데이트와 관련된 의존성을 정확히 설정하고, useEffect 내의 상태 변경을 조건부로 처리했습니다.
- 상태 관리의 복잡성: useReducer, themeReducer을 사용하여 복잡한 상태 변경 로직을 관리함으로써 상태 관리의 복잡성을 줄이고자 했습니다.
- 다양한 화면 크기에 대한 레이아웃 문제: window.matchMedia를 사용하여 화면 크기에 따라 레이아웃을 조정하고 반응형 디자인을 구현했습니다.

### 🌈 개선 과제

- 라우트 핸들러, 탠스택 쿼리로 전환이 필요

### 🌈 팀에 대한 평가

- 10점 만점 중 9점
- 소통 및 회의가 체계적이었고, 팀원분들이 맡은 부분 이상을 잘 해내주셨습니다.

### 🌈 스스로에 대한 평가

- 10점 만점 중 5점
- 구현하고자 했던 부분을 완벽히 끝내지 못했고 개선되어야 할 부분이 남아 아쉽습니다.

### 🌈 부트캠프를 마치며 하고 싶은 말

- 협업에 대해 조금 더 알아가는 시간이 되었고 고생하신 팀원분들 수고 많으셨습니다.

**Reply by slsl2** [reply]
### 🌈 맡은 기능

1) 실시간 채팅 기능
2) 실시간 알림 기능 일부
3) home 페이지, 헤더, 탭메뉴, 기능 일부

### 🌈 사용한 라이브러리와 이유

- zustand 사용해 count 전역 상태 관리 : 코드가 복잡하지 않으며 사용이 편리하고 유지보수에 용이

### 🌈 기술적으로 강조할 부분

- supabase realtime 활용한 실시간 인터렉션
- supabase SQL editor로 안 읽은 메시지 숫자를 반환해주는 함수를 작성해 사용
- useTapScroll 등 재사용이 가능한 코드는 hooks, utils 파일 만들어 재사용

### 🌈 기술적 의사 결정 과정/사유

- 안 읽은 메시지 개수 세는 SQL 작성: 대량 데이터 처리가 필요한데 DB에서 계산해주면 효율적이라고 생각했습니다. DB 변경은 없는데 실시간으로 소통할 수 있고 클라이언트 사이드에서 코드로 가져오는 것보다 간결하고 처리가 빠를 것이라고 판단했습니다.

### 🌈 트러블 슈팅과 해결 과정

- 채팅 목록(chat)과 채팅방(chat/[id] 페이지 데스크탑 구현. 모바일에서는 따로 분리되었던 두 페이지의 UI가 데스크탑에선 같이 보여져야 했던 문제.
채팅 목록에서 채팅을 누르면 목록은 그대로 남아있고 옆에 채팅방이 뜨도록 만들어야 했습니다. 이 부분이 막막해서 며칠을 인스타그램 등의 다른 유사 서비스를 살펴보며, 구글링도 해보며 고민하고 우선은 다른 기능부터 작업하다가, 문득 정말 간단한 방법이 있다는 걸 떠올릴 수 있었습니다. chat 폴더의 layout.tsx에 컴포넌트를 넣는 방법입니다. layout.tsx에 채팅 리스트 컴포넌트를 넣고, 1280px 이상일 경우에만 보이도록 설정해주어서 간단히 해결할 수 있었습니다.
처음 해본다고 어렵게 생각하거나 두려워하지 말고, 기초부터 생각해보면 쉽게 답이 나올 수도 있다는 것을 배웠습니다.

### 🌈 개선 과제

- 맡은 파트 전체적으로 리팩토링

### 🌈 팀에 대한 평가

- 10점 만점 중 9점
- 모두 맡으신 파트 잘 해주셨습니다. 특히 리더님과 부리더님이 맡은 파트 외의 개발적인 부분도, 개발 외적인 일들도 열정을 다해주시고 이끌어주신 덕분에 완성도 높은 프로젝트를 만들 수 있었습니다. 또한 디자이너님과 함께하는 첫 협업이라는 점에서도 어려움이 있을 수 있었지만, 서로 소통이 원만하게 이루어졌다고 느꼈습니다.

### 🌈 스스로에 대한 평가

- 10점 만점 중 5점
- 채팅 구현은 예전부터 해보고 싶었던 기능이었는데 좋은 경험이 되었습니다. 실시간 기능 구현, 깃허브 활용, zustand 사용, SQL작성 등 처음 해보는 게 많았는데 도전해보길 잘 했다고 느끼고 있습니다. 하지만 처음 팀에서 지키자고 했던 컨벤션을 못 지키고 정돈되지 않은 코드를 작성하게 되었고, 맡은 파트 구현 자체도 100% 하지는 못 하고 팀원 분들의 도움을 받아 완성시킬 수 있었어서, 프로젝트는 잘 마무리 되었지만 개인적으로 아쉬움이 남습니다.

### 🌈 부트캠프를 마치며 하고 싶은 말

- 파이널 프로젝트에서 좋은 팀을 만나 많이 배우고 캠프를 잘 마무리 할 수 있었습니다! 고생 많으셨습니다.

**Comment by Captain-Kim** [comment]
## PPT 샘플

<table>
  <tr>
    <th>샘플</th>
    <th>샘플</th>
  </tr>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/17d1b311-f16a-43d5-937c-b1504f9a7631" width="300" height="300"/></td>
    <td><img src="https://github.com/user-attachments/assets/07c25d0d-8cb1-42f2-9f49-97a74ce28647" width="300" height="300"/></td>
  </tr>
</table>

**Reply by Captain-Kim** [reply]
## PPT 구성

1. 표지 1 (웨이팅)
2. 표지 2 (프로젝트 개요)
3. 목차
4. 프로젝트 개요
5. 프로젝트 비즈니스 문제와 해결, 핵심 기능, 기대효과
6. 팀원 소개
7. 와이어프레임1 (모바일1)
8. 와이어프레임2 (모바일2)
9. 와이어프레임3 (PC1)
10. 와이어프레임4 (PC2)
11. 와이어프레임5 (PC3)
12. 와이어프레임6 (PC4)
13. 팀 프로젝트 타임라인 / 일일 일과표
14. 주요 기술 스택
15. 사용한 라이브러리
16. 주요 협업 툴 및 방식
    1. GitHub Organization의 기능을 활용한 협업 : 타임라인, discussions, issues 등
    2. Slack을 활용한 소통
    3. Zep을 활용한 실시간 화상회의
    4. FIgma를 통한 디자인 협업
17. 프로젝트 아키텍쳐
18. ERD 다이어그램
19. API 명세서
20. 팀 컨벤션 (PR, 커밋)
21. MVP 소개
22. 튜터 피드백과 반영 내용
23. UT 결과와 반영 내용
24. 라이트하우스 최적화 지수 분석
25. 개인별 회고
    1. 트러블 슈팅과 기술적 의사결정
    2. 팀 평가 / 개인 평가
    3. 개선과제
    4. 마지막 한 마디

**Comment by Captain-Kim** [comment]
## 🌈 발표 일정

**Section C 18:30 ~ 18:45**

![image](https://github.com/user-attachments/assets/bf2cc9e6-c0a6-4393-9203-2ceecd862492)

**Comment by Captain-Kim** [comment]
![CleanShot 2024-08-19 at 03 34 09](https://github.com/user-attachments/assets/9b2acb44-2372-4cf7-aef8-4d77677d1c82)

**TODO**: 이 부분에 들어갈 내용, 공통으로 했어도 조금 더 메인으로 하셨던 분이 **본인 스스로 어필할 수 있는 내용** 댓글로 적어주세요

**Comment by eunohhh** [comment]
## 개인별 회고록 (오은)

- 어필하고 싶은 부분 부끄러워 마시고 고생하신 만큼 제대로 PR 해서 다 적어 주시고,
- 공통으로 한 기능도 본인이 참여한 내용을 강조해서 전부 적어주세요.
- 트러블 슈팅은 기술적 의사 결정 과정과 사고의 흐름이 잘 보이도록 해결 과정을 상세히 서술해주세요.

### 🌈 맡은 기능

1) 인증 / 인가
2) 회원정보 수정
3) 실시간 알림 기능
4) 온도지수 관련 기능
5) 스토리 관련 기능
6) 이미지 자동생성 기능

### 🌈 사용한 라이브러리와 이유

- cva : 재사용성 높은 공용 컴포넌트를 위해 사용함
- openai : 이미지 자동생성을 위해 사용함
- react-intersection-observer : 무한 스크롤 구현을 위해 사용함
- sharp : 서버측에서 이미지를 webp 로 변환하기 위해 사용함
- react-draggable : 요소 드래그를 구현하기 위해 사용함

### 🌈 기술적으로 강조할 부분

1) 재사용 가능한 컴포넌트 및 커스텀 훅 설계
2) 서버측 프리페치 쿼리
3) 인증/인가 레이어를 context api 로 구현
3) 수파베이스 리얼타임을 활용한 알림 기능을 context api 로 구현

### 🌈 기술적 의사 결정 과정/사유
- nextjs 의 장점을 활용하기 위해 가능한 경우 최대한 서버측에서 프리페치 쿼리 하여, 초기 데이터가 있는 상태로 클라이언트 컴포넌트가 시작되게 하였습니다.
- 서버로부터 prop 을 받아 initialState 를 설정할 수 있는 context api 를 활용하여 전역상태 관리 라이브러리를 사용하지 않고도 상태관리를 효율적으로 할 수 있었습니다.
- 아토믹 패턴을 최대한 활용하여 효율적인 컴포넌트 관리를 하려고 하였습니다. 이 과정에서 cva를 사용하여 최대한 쉽게 재사용 가능한 컴포넌트를 만들려고 하였습니다.

### 🌈 트러블 슈팅과 해결 과정
- pwa 를 구현함에 있어 ios 에서는 안내를 보여주는 방법밖에는 없다는 것을 확인했고, 해결방법이 없음에 따라 제외하게 되었습니다.
- supabase 인증에서 발생하는 Auth session missing! 에러에 대처하기 위해, 이 부분을 라우트 핸들러에서 예외 처리하여 해결할 수 있었습니다.
- 모바일에서 당겨서 새로고침 동작을 막아야 하는 상황이 있었는데 useEffect 에서 documentElement 에 선택적으로 overscrollBehavior='none'을 주어서 해결했습니다.

### 🌈 개선 과제
- 라우트핸들러 사용법, fetch 사용법, 탠스택쿼리 사용법 등에서 통일감 있게 진행되지 못한 부분이 많습니다. 개선이 필요합니다.
- 아토믹 패턴을 가능케 하는 컴포넌트 분리에 있어 팀원간에 컨벤션 통일이 정확히 이루어지지 못한 것 같습니다. 개선이 필요합니다.
- 코드 스플리팅을 통한 사이트 속도 개선이 필요해보입니다.
- seo, web vital 부분에서 부족한 부분이 많습니다. 역시 개선이 필요해보입니다.

### 🌈 팀에 대한 평가

- 10점 만점 중 9점
- 이유는 소통이 잘 되었고 긴밀하게 협업하면서 완성까지 다함께 달려온 것 같습니다.

### 🌈 스스로에 대한 평가

- 10점 만점 중 5점
- 컨벤션, 아토믹 패턴, 퍼널 패턴, 탠스택쿼리 사용, 라이브러리 사용 모든 측면에서 프로젝트가 통일감있게 진행되도록 컨트롤 하지 못한 것 같아서 아쉽습니다.

### 🌈 부트캠프를 마치며 하고 싶은 말
- 모두 그동안 빡빡한 일정 속에 너무나 수고 많으셨습니다.
